src/pudl_archiver/archivers/classes.py

```diff
@@ -49,15 +49,31 @@ class _HyperlinkExtractor(HTMLParser):
 
     def __init__(self):
         """Construct parser."""
-        self.hyperlinks = set()
+        self.hyperlinks = dict()
```

Comment:
I think to fix the test failure you just need to change this to:

```suggestion
        self.hyperlinks = {}
```